great_expectations/checkpoint/checkpoint.py

```diff
@@ -127,6 +126,9 @@ def run(
         run_time = run_time or datetime.datetime.now()
         runtime_configuration = runtime_configuration or {}
         result_format = result_format or runtime_configuration.get("result_format")
+        assert isinstance(
+            result_format, (type(None), str, dict)
+        ), "result_format should either be a string or dict if supplied, or None if not supplied."
 
```

Comment:
```suggestion
        _result_format_types = (type(None), str, dict)
        assert isinstance(
            result_format, result_format_types
        ), f"result_format should either be one of type - {' '.join(_result_format_types)}"

```

Could do something like this 🤷